src/test/java/org/pesmypetcare/webservice/dao/MedicationDaoTest.java

```diff
@@ -0,0 +1,461 @@
+import org.pesmypetcare.webservice.error.DatabaseAccessException;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.ArgumentMatchers.same;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.willThrow;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+public class MedicationDaoTest {
+    private static List<Map<List<String>, Object>> medList;
+    private static MedicationEntity medEntity;
+    private static String owner;
+    private static String petName;
+    private static String date;
+    private static String date2;
+    private static String name;
+    private static String dateName;
+    private static String field;
+    private static int value;
+
+
+    private static final String USERS_KEY = "users";
+    private static final String PETS_KEY = "pets";
+    private static final String MEDICATION_KEY = "medications";
+    private final String EXCECUTION_EXC_MSG = "Should throw DatabaseAccessException when ExecutionException received";
+    private final String INTERRUPTED_EXC_MSG = "Should throw DatabaseAccessException when InterruptedException "
+            + "received";
+    private final String DOCUMENT_NOT_EXISTS_EXC_MSG = "Should throw DatabaseAccessException when med document"
+            + " doesn't exist";
+
+    @Mock
+    private Firestore db;
+    @Mock
+    private CollectionReference usersRef;
+    @Mock
+    private DocumentReference ownerRef;
+    @Mock
+    private CollectionReference petsRef;
+    @Mock
+    private DocumentReference petRef;
+    @Mock
+    private CollectionReference medsRef;
+    @Mock
+    private DocumentReference medRef;
+    @Mock
+    private ApiFuture<QuerySnapshot> futureQuery;
+    @Mock
+    private QuerySnapshot querySnapshot;
+    @Mock
+    private ApiFuture<DocumentSnapshot> futureDocument;
+    @Mock
+    private DocumentSnapshot documentSnapshot;
+    @Mock
+    private List<QueryDocumentSnapshot> medsDocuments;
+    @Mock
+    private Iterator<QueryDocumentSnapshot> it;
+
+    @InjectMocks
+    private MedicationDao medDao = new MedicationDaoImpl();
+
+    @BeforeAll
+    public static void setUp() {
+        medList = new ArrayList<>();
+        medEntity = new MedicationEntity();
+        owner = "usuario_de_pruebas";
+        petName = "linux";
+        date = "26-03-2020";
+        date2 = "26-04-2020";
+        dateName = "26-03-2020%Cloroform";
+        field = "duration";
+        value = 4;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['4' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)